### PR TITLE
New CRA Templates

### PR DESCRIPTION
**Pull request by joebochill**
For PXBLUE-1723

#### Changes proposed in this Pull Request:
- Add create-react-app formatted templates for BLANK (typescript and plain javascript)
- These have been published as alpha packages for testing integration into the CLI (https://github.com/pxblue/pxblue-cli/pull/46)
- **@pxblue/cra-template-blank** & **@pxblue/cra-template-blank-typescript**

> NOTE: This PR is mostly around the js template. The typescript template was created and pushed as the initial commit to this repository. You'll want to look at that too, but you won't see it in the File Diff.

#### Screenshots:
image

### CODE REVIEW

**Comment by joebochill** [comment]
You may want to wait on testing these until I have the PR up for the CLI (it will make it easier to test them as an actual project rather than just some template files)

**Comment by jeffvg** [comment]
The blank template drops of the links to resources when resized down to a width of 599 or less via dev tools.

**Comment by huayunh** [comment]
React Native CLI also accepts templates in a similar manner as CRA. 
https://github.com/react-native-community/cli/blob/master/docs/commands.md#--template-string 

Should we do React and React Native as one repository?

**Comment by huayunh** [comment]
A quick search on Angular reveals that `ng` doesn't support any template flags

**Comment by joebochill** [comment]
> The blank template drops of the links to resources when resized down to a width of 599 or less via dev tools.

This is by design. I chopped them off so that the demo included an example of using media queries and responsiveness. Also I didn't feel like spending a ton of time making them look good on really small screens 😆 

That being said, we can change it if we want to.

**Comment by joebochill** [comment]
> React Native CLI also accepts templates in a similar manner as CRA.
> https://github.com/react-native-community/cli/blob/master/docs/commands.md#--template-string
> 
> Should we do React and React Native as one repository?

We certainly could. It depends on how many / how complicated they are going to be. It seems like for most of what we do we have the option to 1) put everything in one repository or 2) divide things up by framework.

**Comment by joebochill** [comment]
> A quick search on Angular reveals that `ng` doesn't support any template flags

I think the angular equivalent is 'schematics'

File: blank-typescript/template/src/App.tsx [hunk 1]
```diff
@@ -54,8 +58,8 @@ export const App = (): JSX.Element => {
     return (
         <div className={classes.pageBackground}>
             <AppBar position={'fixed'}>
-                <Toolbar>
-                    <IconButton color={'inherit'} edge={'start'}>
+                <Toolbar className={classes.toolbar}>
+                    <IconButton color={'inherit'} edge={'start'} style={{ marginRight: theme.spacing(4) }}>
```
Comment (huayunh) [inline blank-typescript/template/src/App.tsx:62]:
```suggestion
                    <IconButton color={'inherit'} edge={'start'} style={{ marginRight: theme.spacing(3) }}>
```

image

File: blank/template/public/manifest.json [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+        {
+            "src": "logo512.png",
+            "type": "image/png",
+            "sizes": "512x512"
+        }
+    ],
+    "start_url": ".",
+    "display": "standalone",
+    "theme_color": "#000000",
+    "background_color": "#ffffff"
```
Comment (huayunh) [inline blank/template/public/manifest.json:24]: Do we want to change manifest to use pxb colors instead?

Docit is using
```json
  "theme_color": "#007bc1",
  "background_color": "#eef0f0"
```

**Review by huayunh — APPROVED** [review]
Shall we set up CircleCI for lint & prettier?

File: blank-typescript/template/src/App.tsx (hunk 1 above)
Comment (huayunh) [reply blank-typescript/template/src/App.tsx:62]: Same for the js side

Comment (joebochill) [reply blank-typescript/template/src/App.tsx:62]: mm, i thought it looked big...didn't account for the 8px padding on the icon.

**Pull request merged**